app/src/main/java/org/simple/clinic/settings/SettingsEffectHandler.kt

```diff
@@ -65,4 +68,16 @@ class SettingsEffectHandler @AssistedInject constructor(
           }
     }
   }
+
+  private fun checkAppUpdateAvailability(
+      appUpdateAvailability: CheckAppUpdateAvailability
+  ): ObservableTransformer<CheckAppUpdateAvailabilityEffect, SettingsEvent> {
+    return ObservableTransformer { effectStream ->
+      effectStream
+          .flatMap {
+            appUpdateAvailability.listen()
```

Comment:
One thing to note is that the `CheckAppUpdateAvailability#listen` method will only emit if the version on the Play Store is ahead of the installed version by a certain value. This was implemented because we didn't want to show a pop-up to users every time they open the app.

However, for this specific screen, it might be better to show the button if there is an update available, regardless of how close to the current app version it is.

If this is the case, we might have to modify `CheckAppUpdateAvailability` to support this use case.

Reply:
Added a method `listenAllUpdates`